guide/src/class/thread-safety.md

```diff
@@ -0,0 +1,53 @@
+# `#[pyclass]` thread safety
+
+Python objects are freely shared between threads by the Python interpreter. This means that:
+- there is no control which thread might eventually drop the `#[pyclass]` object, meaning `Send` is required.
+- multiple threads can potentially be reading the `#[pyclass]` data simultaneously, meaning `Sync` is required.
+
+This section of the guide discusses various datastructures which can be used to make types satisfy these requirements.
```

Comment:
Can we use "data structures" instead? Not sure if this is a british/american english thing, but we seem to do so already:

```
goldbaum at Nathans-MBP in ~/Documents/pyo3 on gil-disabled-build-config!
± rg datastructure

goldbaum at Nathans-MBP in ~/Documents/pyo3 on gil-disabled-build-config!
± rg "data structure"
guide/src/free-threading.md
67:If you are sure that all data structures exposed in a `PyModule` are
110:provide mutable data structures relied on the GIL to make interior mutability

README.md
211:- [mocpy](https://github.com/cds-astro/mocpy) _Astronomical Python library offering data structures for describing any arbitrary coverage regions on the unit sphere._
```

Reply:
I think you are correct on "data structure" and also "thread-safe" below.